dvc/output/__init__.py

```diff
@@ -50,87 +52,53 @@
 TAGS_SCHEMA = {schema.Optional(str): CHECKSUM_SCHEMA}
 
 SCHEMA = CHECKSUM_SCHEMA.copy()
-SCHEMA[OutputBase.PARAM_PATH] = str
-SCHEMA[schema.Optional(OutputBase.PARAM_CACHE)] = bool
-SCHEMA[schema.Optional(OutputBase.PARAM_METRIC)] = OutputBase.METRIC_SCHEMA
-SCHEMA[schema.Optional(OutputBase.PARAM_TAGS)] = TAGS_SCHEMA
-SCHEMA[schema.Optional(OutputBase.PARAM_PERSIST)] = bool
+SCHEMA[OutputBASE.PARAM_PATH] = str
+SCHEMA[schema.Optional(OutputBASE.PARAM_CACHE)] = bool
+SCHEMA[schema.Optional(OutputBASE.PARAM_METRIC)] = OutputBASE.METRIC_SCHEMA
+SCHEMA[schema.Optional(OutputBASE.PARAM_TAGS)] = TAGS_SCHEMA
+SCHEMA[schema.Optional(OutputBASE.PARAM_PERSIST)] = bool
 
 
-def _get(stage, p, info, cache, metric, persist=False, tags=None):
-    parsed = urlparse(p)
+def Output(stage, info):
```

Comment:
the same - super confusing, not clear why do we need it if have a BASE class ... naming for all these BASEs is pretty funky, feels like it would be better to just name them Output, Dependency, Remote, etc ... and put some static helpers to initialize specific subclasses if needed

Reply:
why not use a regular mechanism for passing dicts around - **kwargs?

Reply:
See `deps/outs: use factory-like class` commit.